package/gluon-core/files/lib/gluon/upgrade/200-wireless

```diff
@@ -10,11 +10,21 @@ if not sysconfig.gluon_version then
   uci:delete_all('wireless', 'wifi-iface')
 end
 
+local function get_channel(radio, config)
+  if uci:get_first('gluon-core', 'wireless', 'preserve_channel_' .. radio) and sysconfig.gluon_version and uci:get('wireless', radio) then
```

Comment:
This should not refer to sysconfig.gluon_version. Just return the current setting if there is a setting and preserve_channel is set; in any other case, return the default setting.